server/ngwmn/services/ngwmn.py

```diff
@@ -309,3 +310,179 @@ def get_features(latitude, longitude, service_root=SERVICE_ROOT):
+
+    json_txt = resp.text
+    stats = json.loads(json_txt)
+    stats['IS_FETCHED'] = 'Y'
+
+    app.logger.debug(stats)
+
+    return stats
+
+
+def get_statistics(agency_cd, site_no):
+    """
+    Call ngwmn_cache for site statistics data.
+
+    :param agency_cd: string agency code
+    :param site_no: alphanumeric site number
+    :returns overall and monthly statistics
+
+    SAMPLE overall
+    {
+      "CALC_DATE": "2018-07-04",
+      "LATEST_VALUE": "213.84",
+      "MAX_DATE": "2018-04-10T13:18:00-06:00",
+      "AGENCY_CD": "USGS",
+      "MEDIAN_VALUE": "217.82",
+      "RECORD_YEARS": "31.6",
+      "MAX_VALUE": "181.7",
+      "MEDIATION": "BelowLand",
+      "MIN_VALUE": "232.96",
+      "SAMPLE_COUNT": "7456",
+      "SITE_NO": "353945105574502",
+      "MIN_DATE": "1986-09-26T12:00:00",
+      "LATEST_PCTILE": "0.78606",
+      "IS_RANKED": "Y"
+    }
+    SAMPLE monthly
+    {
+      "3": {
+        "RECORD_YEARS": "23",
+        "P50": "216.54",
+        "P25": "222.72",
+        "P10": "226.24",
+        "P50_MAX": "184.17",
+        "P90": "188.72",
+        "P75": "212.37",
+        "AGENCY_CD": "USGS",
+        "SAMPLE_COUNT": "673",
+        "MONTH": "3",
+        "SITE_NO": "353945105574502",
+        "P50_MIN": "227.46"
+      },
+      "1": {
+        "RECORD_YEARS": "20",
+        "P50": "218.28",
+        "P25": "225.50",
+        "P10": "227.03",
+        "P50_MAX": "184.14",
+        "P90": "190.49",
+        "P75": "214.81",
+        "AGENCY_CD": "USGS",
+        "SAMPLE_COUNT": "608",
+        "MONTH": "1",
+        "SITE_NO": "353945105574502",
+        "P50_MIN": "227.89"
+      }
+    }
+    """
+    site_info = monthly = {'IS_FETCHED': 'N'}
+
+    overall = get_statistic(agency_cd, site_no, 'wl-overall')
+    if overall['IS_FETCHED'] == 'Y':
+        site_info = get_statistic(agency_cd, site_no, 'site-info')
+        if overall['IS_RANKED'] == 'Y':
+            monthly = get_statistic(agency_cd, site_no, 'wl-monthly')
+
+    # handle to potential fetch fail with default
+    stats = {
+        'alt_datum': 'unknown',
+        'calc_date': 'unknown',
+        'overall': [],
+        'monthly': []
+    }
+
+    if overall['IS_FETCHED'] == 'Y':
+        alt_datum_cd = ''
+        if site_info['IS_FETCHED'] == 'Y':
+            alt_datum_cd = site_info['altDatumCd']
+
+        if overall['MEDIATION'] == 'BelowLand':
+            stats['alt_datum'] = 'Depth to water, feet below land surface'
+        else:
+            stats['alt_datum'] = 'Water level in feet relative to ' + alt_datum_cd
+
+        stats['calc_date'] = overall['CALC_DATE']
+
+        stats['overall'] = [
+            overall['MIN_VALUE'],
+            overall['MEDIAN_VALUE'],
+            overall['MAX_VALUE'],
+            overall['MIN_DATE'],
+            overall['MAX_DATE'],
+            overall['SAMPLE_COUNT'],
+            overall['RECORD_YEARS'],
+            overall['LATEST_VALUE'],
+            overall['LATEST_PCTILE']
+        ]
```

Comment:
Sorry for not suggesting this earlier, but it would be nice here if this (and the monthly stats, below) were returned as dictionaries... that would allow the template to not need to know the order of the fields in the array.

Also, I'd suggest returning the dates as `datetime.date` objects, and then in the template, doing locale formatting on them. Not a big deal though.

Reply:
If I do that then I cannot loop over the entries in the template.

Reply:
I think it's better to be explicit in the template rather than looping over an array. Also allows reordering of the columns if we desire to occur only in the template

Reply:
I considered that. In fact, the original/legacy page do it that way. I chose the loop to match other implementations in the template.

Reply:
Looking at it again, I am not sure how I misinterpreted that. I would have coded it to match that pattern had I seen it that way. Somehow, I saw the existing as a loops over columns not rows.

Reply:
Next question. The JSON returns "ALL CAPS" entries. Should I just pass that to the template or would you both prefer some other non-cap format?

Reply:
Yes, I'd prefer switching to lower case and making the attribute names "Pythonic". It reinforces that it's not verbatim what the service returns.